tools/operator/fabricconfiguration/configtx.go

```diff
@@ -0,0 +1,933 @@
+		if err != nil {
+			return configtx.MSP{}, fmt.Errorf("parsing peer ou identifier cert: %v", err)
+		}
+
+		adminOUIdentifierCert, err = parseCertificateFromBytes(fabricMSPConfig.FabricNodeOus.AdminOuIdentifier.Certificate)
+		if err != nil {
+			return configtx.MSP{}, fmt.Errorf("parsing admin ou identifier cert: %v", err)
+		}
+
+		ordererOUIdentifierCert, err = parseCertificateFromBytes(fabricMSPConfig.FabricNodeOus.OrdererOuIdentifier.Certificate)
+		if err != nil {
+			return configtx.MSP{}, fmt.Errorf("parsing orderer ou identifier cert: %v", err)
+		}
+
+		nodeOUs = membership.NodeOUs{
+			Enable: fabricMSPConfig.FabricNodeOus.Enable,
+			ClientOUIdentifier: membership.OUIdentifier{
+				Certificate:                  clientOUIdentifierCert,
+				OrganizationalUnitIdentifier: fabricMSPConfig.FabricNodeOus.ClientOuIdentifier.OrganizationalUnitIdentifier,
+			},
+			PeerOUIdentifier: membership.OUIdentifier{
+				Certificate:                  peerOUIdentifierCert,
+				OrganizationalUnitIdentifier: fabricMSPConfig.FabricNodeOus.PeerOuIdentifier.OrganizationalUnitIdentifier,
+			},
+			AdminOUIdentifier: membership.OUIdentifier{
+				Certificate:                  adminOUIdentifierCert,
+				OrganizationalUnitIdentifier: fabricMSPConfig.FabricNodeOus.AdminOuIdentifier.OrganizationalUnitIdentifier,
+			},
+			OrdererOUIdentifier: membership.OUIdentifier{
+				Certificate:                  ordererOUIdentifierCert,
+				OrganizationalUnitIdentifier: fabricMSPConfig.FabricNodeOus.OrdererOuIdentifier.OrganizationalUnitIdentifier,
+			},
+		}
+	}
+
+	return configtx.MSP{
+		Name:                          fabricMSPConfig.Name,
+		RootCerts:                     rootCerts,
+		IntermediateCerts:             intermediateCerts,
+		Admins:                        adminCerts,
+		RevocationList:                revocationList,
+		OrganizationalUnitIdentifiers: ouIdentifiers,
+		CryptoConfig: membership.CryptoConfig{
+			SignatureHashFamily:            fabricMSPConfig.CryptoConfig.SignatureHashFamily,
+			IdentityIdentifierHashFunction: fabricMSPConfig.CryptoConfig.IdentityIdentifierHashFunction,
+		},
+		TLSRootCerts:         tlsRootCerts,
+		TLSIntermediateCerts: tlsIntermediateCerts,
+		NodeOUs:              nodeOUs,
+	}, nil
+}
+
+func getMspConfig(dir string, ID string) (*msp.MSPConfig, error) {
+	cacertDir := filepath.Join(dir, cacerts)
+	admincertDir := filepath.Join(dir, admincerts)
+	intermediatecertsDir := filepath.Join(dir, intermediatecerts)
+	crlsDir := filepath.Join(dir, crlsfolder)
+	configFile := filepath.Join(dir, configfilename)
+	tlscacertDir := filepath.Join(dir, tlscacerts)
+	tlsintermediatecertsDir := filepath.Join(dir, tlsintermediatecerts)
+
+	cacerts, err := getPemMaterialFromDir(cacertDir)
+	if err != nil || len(cacerts) == 0 {
+		return nil, errors.WithMessagef(err, "could not load a valid ca certificate from directory %s", cacertDir)
```

Comment:
Don't use the error's package, use `fmt.Errorf` to construct errors. There are several places we use it today

Reply:
Fixed